pkg/apis/github/git.go

```diff
@@ -17,14 +17,24 @@ func (g *Github) DeleteRef(repository, branchName string) error {
 }
 
 // CreateRef creates a new ref (GitHub branch) in a specified GitHub repository,
-// that will be based on the latest commit from a specified branch name
-func (g *Github) CreateRef(repository, baseBranchName, newBranchName string) error {
+// that will be based on the commit specified with sha. If sha is not specified
+// the latest commit from base branch will be used.
+func (g *Github) CreateRef(repository, baseBranchName, sha, newBranchName string) error {
 	ctx := context.Background()
 	ref, _, err := g.client.Git.GetRef(ctx, g.organization, repository, fmt.Sprintf("heads/%s", baseBranchName))
 	if err != nil {
 		return fmt.Errorf("error when getting the base branch name '%s' for the repo '%s': %+v", baseBranchName, repository, err)
 	}
+
 	ref.Ref = github.String(fmt.Sprintf("heads/%s", newBranchName))
+
+	if sha != "" {
+		ref.Object.SHA = &sha
+
+		refURL := fmt.Sprintf("%s/%s", (*ref.Object.URL)[:strings.LastIndex(*ref.Object.URL, "/")], sha)
+		ref.Object.URL = &refURL
```

Comment:
is this part necessary?
according to [docs](https://docs.github.com/en/rest/git/refs?apiVersion=2022-11-28#create-a-reference) only the ref name and sha is required

Reply:
It is not necessary right now but it keeps the consistency of the Reference struct. May be too defensive but I would still suggest to keep it. Wdyt?

Reply:
Removed.